public/app/features/explore/ShortLinkButtonMenu.tsx

```diff
@@ -0,0 +1,99 @@
+import { css } from '@emotion/css';
+import React, { useState } from 'react';
+
+import { GrafanaTheme2, urlUtil } from '@grafana/data';
+import { Stack } from '@grafana/experimental';
+import { reportInteraction } from '@grafana/runtime';
+import { useStyles2, ToolbarButton, Dropdown, Menu } from '@grafana/ui';
+import { t } from 'app/core/internationalization';
+import { createAndCopyShortLink } from 'app/core/utils/shortLinks';
+import { useSelector } from 'app/types';
+
+import { getUrlStateFromPaneState } from './hooks/useStateSync';
+import { selectPanes } from './state/selectors';
+
+export function ShortLinkButtonMenu() {
+  const styles = useStyles2(getStyles);
+  const panes = useSelector(selectPanes);
+  const [isOpen, setIsOpen] = useState(false);
+  const onCopyShortLink = (url?: string) => {
+    createAndCopyShortLink(url || global.location.href);
+    reportInteraction('grafana_explore_shortened_link_clicked');
+  };
+
+  const MenuActions = () => {
+    return (
+      <Menu>
+        <Menu.Item
+          key="copy-link"
+          label="Copy shortened URL"
+          icon="link"
+          onClick={() => onCopyShortLink()}
+          className={styles.menuItem}
+        />
+        <Menu.Item
+          key="copy-link-abs-time"
+          label="Copy absolute time URL"
+          icon="clock-nine"
+          description="Locks the current time for consistent viewing"
+          onClick={() => {
+            const exploreIds = Object.keys(panes);
+            const urlStates = Object.values(panes)
+              .map((pane, i) => {
+                if (pane) {
+                  const urlState = getUrlStateFromPaneState(pane);
+                  urlState.range = {
+                    to: pane.range.to.valueOf().toString(),
+                    from: pane.range.from.valueOf().toString(),
+                  };
+                  return JSON.stringify({ [exploreIds[i]]: { ...urlState } });
+                } else {
+                  return undefined;
+                }
+              })
```

Comment:
you can achieve the same by using `Object.entries`

```suggestion
            const urlStates = Object.entries(panes)
              .map(([exploreId, pane], i) => {
                if (!pane) {
                  return;
                }

                const urlState = getUrlStateFromPaneState(pane);
                urlState.range = {
                  to: pane.range.to.valueOf().toString(),
                  from: pane.range.from.valueOf().toString(),
                };
                return JSON.stringify({ [exploreId]: { ...urlState } });
              })
```

Reply:
fixed!